libs/portals/admin/ids-admin/src/screens/Client/components/AddPermissions/AddPermissions.tsx

```diff
@@ -0,0 +1,243 @@
+  } = useClient()
+  const fetcher = useFetcher()
+  const [selectedTenant, setSelectedTenant] = useState<Option>()
+  const [availableTenants, setAvailableTenants] = useState<AuthTenants>([])
+  const [availableScopes, setAvailableScopes] = useState<AuthAdminScopes>([])
+
+  const [
+    getAvailableScopesQuery,
+    { data: { authAdminScopes } = { authAdminScopes: undefined }, loading },
+  ] = useGetAvailableScopesLazyQuery({
+    fetchPolicy: 'network-only',
+  })
+
+  const [selected, setSelected] = useState<
+    Map<string, AuthAdminClientAllowedScope>
+  >(new Map())
+
+  useEffect(() => {
+    if (fetcher.state === 'idle' && !fetcher.data) {
+      fetcher.load(IDSAdminPaths.IDSAdmin)
+    }
+  }, [fetcher])
+
+  useEffect(() => {
+    if (fetcher.data) {
+      // When fetcher has loaded data we filter available tenants for the selected environment
+      const availableTenants = (fetcher.data as AuthTenants).filter((tenant) =>
+        tenant.availableEnvironments.find(
+          (availableEnvironment) => availableEnvironment === environment,
+        ),
+      )
+      // Populate the available tenants state for dropdown values
+      setAvailableTenants(availableTenants)
+
+      // Find the selected tenant based on the tenant router param
+      const selectedTenant = availableTenants.find((t) => t.id === tenant)
+
+      if (selectedTenant) {
+        setSelectedTenant(formatOption(selectedTenant, locale))
+        getAvailableScopesQuery({
+          variables: {
+            input: {
+              tenantId: selectedTenant.id,
+            },
+          },
+        })
+      }
+    }
+  }, [fetcher.data, locale])
+
+  useEffect(() => {
+    if (authAdminScopes?.data) {
+      // When scope data is queried we need to filter it for the selected environment
+      // and remove already added or allowed scopes.
+      setAvailableScopes(
+        authAdminScopes.data
+          .map((scope) =>
+            scope.environments.find((e) => e.environment === environment),
+          )
+          .filter(isDefined)
+          .filter(
+            (scope) =>
+              scope &&
+              ![...addedScopes, ...(allowedScopes ?? [])].find(
+                (s) => s.name === scope.name,
+              ),
+          ),
+      )
+    }
+  }, [authAdminScopes?.data])
+
+  // Add the selected scopes to the addedScopes array for
+  const handleAdd = () => {
+    // Add the new scopes to the permissions array
+    onAdd([...selected.values()])
+
+    // Reset the selected scopes
+    setSelected(new Map())
+
+    // Close the modal
+    onClose()
+  }
+
+  // Handle checkbox change
+  const onChange = (value: AuthAdminClientAllowedScope) => {
+    if (selected.has(value.name)) {
+      selected.delete(value.name)
+    } else {
+      selected.set(value.name, value)
+    }
+    setSelected(new Map(selected))
+  }
+
+  const handleTenantChange = (opt: Option) => {
+    setSelectedTenant(opt)
+
+    getAvailableScopesQuery({
```

Comment:
nit: since this query fn is called twice you could extract it into a helper fn. Something like
```
  const fetchAvailableScopes = (tenantId: string) =>
    getAvailableScopesQuery({
      variables: {
        input: {
          tenantId,
        },
      },
    })
```